tests/ecmascript/test-bi-object-assign.js

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Object.assign() (E7 19.1.2.1)
+ *  http://www.ecma-international.org/ecma-262/7.0/#sec-object.assign
+ */
+
+/*===
+basic tests
+true
+maggie Kittycow Machel Fatty Whale 812 1208 9001
+0,1,2,pig,cow,ape,whale
+maggie 1208 9001 Vegeta
+cow,pig,ape,saiyan
+undefined
+this belongs here
+undefined
+prop1
+get
+set 812
+===*/
+
+function basicTest() {
+    var obj, src, dest;
+
+    // If no coercion is needed, target object is returned (i.e. not a copy).
+    obj = {};
+    dest = Object.assign(obj, { test: "test" });
+    print(dest === obj);
+
+    // Properties are assigned in [[OwnPropertyKeys]] order, moving from left
+    // to right in the argument list:
+    //     1. Array indices (sequential, ascending)
+    //     2. String keys (creation order)
+    //     3. Symbol keys (creation order)
+    // This behavior is easy to test for by assigning from multiple sources to
+    // an empty object and then calling Reflect.ownKeys() on the result.
```

Comment:
This comment doesn't seem right: if Object.assign() worked in the incorrect order, using Reflect.ownKeys() wouldn't reveal that because *it* would reorder the keys?

So I guess the only way to detect the correct order is to have printing getters with array indices, non-array indices etc, and to see that they're called in the expected order?